README.md
# Antwerp UI - React Components

## Introduction

Antwerp City Platform as a Service User Interface (Antwerp UI) is a **component interface library** for building user interfaces and responsive web apps. It's designed to provide developers with functionality and UI/UX patterns that matches the Antwerpen styleguide.

## Ecosystem

This library is part of [Antwerp UI][antwerp-ui].

| Name              | Framework  | Status  |
| ----------------- | ---------- | ------- |
| Javascript        | ES5+       | [![antwerp-ui-js-github]][antwerp-ui-js] |
| Angular           | Angular 6+ | [![antwerp-ui-angular-github]][antwerp-ui-angular] |
| React             | React 16+  | [![antwerp-ui-react-github]][antwerp-ui-react] |

## Getting Started

```sh
npm install @acpaas-ui/react-components
```

Import the core branding to style the components:

```scss
@import url("https://cdn.antwerpen.be/core_branding_scss/5.0.0/main.min.css");
```

You can also add a branding-compatible [flexbox grid layout system][flexboxgrid]:

```scss
@import url('https://cdn.antwerpen.be/core_flexboxgrid_scss/2.0.0/flexboxgrid.min.css');
```

Note that you are free to use your own grid, as long as it meets our branding guidelines.

> To learn more about using the branding look & feel, check the [core branding guide][branding-core-guide]. Note that the [core branding][branding-core] contains the Antwerp logo and fonts, which are subject to a usage restriction. For use outside of apps for the city of Antwerp you can use [acpaas branding][branding-acpaas] instead.

Check out the [live examples and API docs](https://digipolisantwerp.github.io/antwerp-ui_react/) to learn more.

## Questions

For questions and support please ask a question on the [#antwerp-ui-react slack channel][antwerp-ui-react-slack]. If you're not yet a member of our DigAnt Café slack community, you can easily [join here][digantcafe-slack].

If you stumble across a bug or missing feature, feel free to [report an issue][antwerp-ui-react-issues]. Please fill out the issue template completely when [opening an issue][antwerp-ui-react-issues]. This will help us get to your issue sooner.

## Changelog

Detailed changes for each release are documented in the [changelog](./CHANGELOG.md).

## Contributing

Your contributions are most welcome as pull requests, both code changes and documentation updates. However, to keep a high quality standard, please make sure to read the [Contributing Guide](./CONTRIBUTING.md) before making a pull request.

Thank you to all the people [who already contributed][antwerp-ui-react-contributors] to Antwerp UI!

## Support

Jasper Van Proeyen (<jasper.vanproeyen@digipolis.be>)

## License

[MIT](./LICENSE.md)

Copyright (c) 2018-present, Digipolis

<!-- Generic Links -->
[antwerp-ui]: https://antwerp-ui.digipolis.be
[antwerp-ui-react-slack]: https://digantcafe.slack.com/messages/CDGCW8U30/
[flexboxgrid]: https://github.com/a-ui/core_flexboxgrid_scss
[digantcafe-slack]: https://digantcafe-slack.digipolis.be

<!-- Github links -->

<!-- Github URL -->
[antwerp-ui-js]: https://github.com/digipolisantwerp/antwerp-ui_js
[antwerp-ui-angular]: https://github.com/digipolisantwerp/antwerp-ui_angular
[antwerp-ui-react]: https://github.com/digipolisantwerp/antwerp-ui_react
[antwerp-ui-react-issues]: https://github.com/digipolisantwerp/antwerp-ui_react/issues
[antwerp-ui-react-contributors]: https://github.com/digipolisantwerp/antwerp-ui_react/graphs/contributors
[branding-core]: https://github.com/a-ui/core_branding_scss
[branding-core-guide]: https://a-ui.github.io/core_branding_scss/
[branding-acpaas]: https://github.com/a-ui/acpaas_branding_scss

<!-- Github Version Badge -->
[antwerp-ui-angular-github]: https://img.shields.io/github/package-json/v/digipolisantwerp/antwerp-ui_angular.svg
[antwerp-ui-react-github]: https://img.shields.io/github/package-json/v/digipolisantwerp/antwerp-ui_react.svg
[antwerp-ui-js-github]: https://img.shields.io/github/package-json/v/digipolisantwerp/antwerp-ui_js.svg


CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/),
and this project adheres to [Semantic Versioning](https://semver.org/).

## [Unreleased]

### Fixed
- `Switch` - not passing correct value.

## v6.4.1 - 2022-08-31

### Fixed
- `Table` Fixed datepicker filter value

## v6.4.0 - 2022-07-01

### Added
- Added support for React 18
- `Form` - Added placeholder option for datepicker
- `Form` - Added description option for select
- `Form` - Added error description option for select
- `Form` - Added error description option for datepicker

### Fixed
- Made sure no extra space is created when flyout is closed


## v6.3.1 - 2022-01-24

### Fixed
- `Table` - export table component


## v6.3.0 - 2022-01-1ß

### Added
- `Table` - added table component


## v6.2.1 - 2021-12-13

### Fixed
- `Autocomplete` - prevent crash when going to index 0


## v6.2.0 - 2021-12-07

### Added
- `Modal` - added option to use custom component as confirm and deny button


## v6.1.0 - 2021-11-25

### Fixed
- `Select` - removed value from internal state
- `Autocomplete` - removed defaultValue from internal state

### Changed
- `Autocomplete` Add disabled option to items array


## v6.0.6 - 2021-10-08

### Fixed
- `Select` - fixed defaultValue not working
- `Icon` - fixed document reference


## v6.0.5 - 2021-10-07

### Fixed
- `Autocomplete` - removed focus state from fake input field


## v6.0.4 - 2021-09-07

### Fixed
- Fixed an issue with hard peer dependencies


## v6.0.3 - 2021-09-02

### Changed
- Removed unnecessary dependencies, moved `react` and `react-dom` to peer dependencies

### Fixed
- `Form/Checkbox` - no longer becomes disabled when no id is given (fixes [#248](https://github.com/digipolisantwerp/antwerp-ui_react/issues/248))
- `Form/Datepicker` - label selects datepicker even when no id is given
- `Form/Datepicker` - cleaned up document event listeners (fixes [#247](https://github.com/digipolisantwerp/antwerp-ui_react/issues/247))
- `Calendar` headers now show monday first for `en` locale (fixes [#246](https://github.com/digipolisantwerp/antwerp-ui_react/issues/246)


## v6.0.2 - 2021-06-08

### Fixed
- Fixed an issue caused by babelHelpers that weren't included in the build


## v6.0.1 - 2021-06-07 [BORKED]

### Fixed
- Fixed the styleguide not building correctly

### Changed
- Optimized build


## v6.0.0 - 2021-06-02

### Changed
- `Icon` [BREAKING] FontAwesome was replaced by support for Streamline icons.
- `Switch` [BREAKING] Changed the way the Switch component works
- `Toggle` [BREAKING] Changed the way the Toggle component works
- `Button` [BREAKING] Removed 'alt' prop as it isn't valid
- Antwerp UI

### Fixed
- `Accordion` Improved accessibility
- `Accordion` Improved accessibility
- `Calendar` Improved accessibility
- `Datepicker` Improved accessibility
- `Modal` Improved accessibility
- `Slider` Improved accessibility

### Removed
- `Subfooter` Please use the metanavigation instead


## v5.0.6 - 30-03-2021

### Fixed
- `Autocomplete` Fixed styling not being included in the build


## v5.0.5 - 22-03-2021

### Fixed
- `Autocomplete` Fixed a missing `FlyoutContent` import
- `Autocomplete` Fixed component not showing in styleguide


## v5.0.4 - 16-03-2021

### Fixed
- `Toggle` Final fix for the toggle's mixed checked state
- Optimized published package size


## v5.0.3 - 11-03-2021

### Fixed
- Fixed styleguide which was showing too many (sub) components
- `Autocomplete` Fixed state error
- `Footer` Fixed double package name for SubFooter
- `Form/TelephoneNumber` Fixed minor styling issues
- `Form/TextField` Fixed an issue with class names
- `Handle` Replaced deprecated componentWillReceiveProps
- `Toggle` Fixed an issue where the toggle's checked state could still get mixed up

### Changed
- Upgraded to the latest core branding (v4.3.1)
- Updated Styleguidist
- `Form/Select` Improved inner workings
- `Form/TelephoneNumber` Fixed minor styling issues


## v5.0.2 - 23-02-2021

### Fixed
- `Form/TelephoneNumber` Fix infinite loop when setting initialValue


## v5.0.1 - 19-02-2021

### Fixed
- `Form/TelephoneNumber` Fix border & spacing


## v5.0.0 - 19-02-2021

### Changed
- `Input` [BREAKING] We had to change the input's API to make it more consistent. This means that for some components 'type' had to be changed to 'size', in order to also support the standard 'type' attribute.

### Added
- `Form/TelephoneNumber` Added new telephone number component


## v4.9.0 - 18-02-2021

### Added
- `Form/TextField` Support addons (addon-left, addon-right)
- `Form/TextField` Pass type prop to child input
- `Form/TextField` Allow for setting className on wrapper

### Fixed
- `Autocomplete` Fixed a bug where search results were not correctly parsed
- `Toggle` Fixed an issue where the toggle's checked state could get mixed up


## v4.8.2 - 11-12-2020

### Fixed
- `Form/Datepicker` Fixed order of the week days
- `Form/TextField` Fixed ability to pass unexpected props
- `General` Fixed changelog


## v4.8.1 - 25-11-2020

### Fixed
- `Autocomplete` Fixed an issue with tabIndex


## v4.8.0 - 20-11-2020

### Added
- `RadioGroup` Added support for RadioOption key attribute
- `RadioGroup` Added support for multiple radiogroups with same values

### Fixed
- `Pagination` Fixed Pagination not rerendering when `itemsPerPage` or `totalValues` change
- `Autocomplete` Fixed an issue where an item sometimes wasn't selected
- `Autocomplete` Improved overall accessibility


## v4.7.0 - 05-10-2020

### Added
- `Textarea` Added an option to pass `onInvalidCapture` as an attribute.
- `TextField` Added support for colored description based on the state of the TextField

### Fixed
- `Checkbox` Use checked instead of defaultChecked


## v4.6.1 - 22-09-2020

### Fixed
- `Textarea` Fixed unit tests
- `Switch` Fixed unit tests
- `Switch` Fixed updating inner state on props change

styleguide.config.js
const path = require('path');
const pjson = require('./package.json');

// Fix GitHub Pages
let basePath = '';
if (process.env.NODE_ENV === 'production') {
  basePath = 'https://digipolisantwerp.github.io/antwerp-ui_react';
}

module.exports = {
  components: 'packages/*/src{/component,}/!(index)*.{js,jsx}',
  getExampleFilename(componentPath) {
    const basePath = componentPath.split('src/')[0];
    return basePath + 'Readme.md';
  },
  compilerConfig: {
    transforms: {
      asyncAwait: false,
      moduleImport: false,
    }
  },
  ignore: [
    path.resolve(__dirname, 'packages/**/models/*.js'),
    path.resolve(__dirname, 'packages/**/types.js'),
    path.resolve(__dirname, 'packages/calendar/src/utils/aria.js'),
    path.resolve(__dirname, 'packages/form/src/TelephoneNumber/TelephoneNumber.const.js'),
    path.resolve(__dirname, 'packages/user-menu/src/UserNavigation/UserNavigationUrls.js'),
    '**/*.spec.js'
  ],
  skipComponentsWithoutExample: true,
  getComponentPathLine: (componentPath) => {
    const name = path.basename(componentPath).replace(/.jsx?$/g, '');
    return `import { ${name} } from '@acpaas-ui/react-components';`;
  },
  dangerouslyUpdateWebpackConfig(webpackConfig, env) {
    // WARNING: inspect Styleguidist Webpack config before modifying it, otherwise you may break Styleguidist
    console.log('Environment: ', process.env.NODE_ENV);
    return webpackConfig;
  },
  styleguideDir: 'docs',
  title: 'Antwerp UI React components',
  usageMode: 'collapse',
  webpackConfig: require('./webpack.config.js'),
  version: `v${pjson.version}`,
  template: {
    context: '',
    favicon: 'https://cdn.antwerpen.be/core_branding_favicons/citizens/favicon.ico',
    lang: 'en',
    head: {
      raw: '<base href="' + basePath + '/">',
      links: [
        {
          rel: 'stylesheet',
          href: 'https://cdn.antwerpen.be/core_branding_scss/5.0.0/main.min.css'
        }
      ],
      scripts: [
        {
          src: "https://cdn.jsdelivr.net/npm/regenerator-runtime@0.13.7/runtime.min.js",
          type: "text/javascript"
        }
      ]
    }
  },
  // styleguideComponents: {
    // Logo: path.join(__dirname, 'styleguide/components/Logo'),
    // SectionsRenderer: path.join(__dirname, 'styleguide/components/SectionsRenderer'),
    // StyleGuideRenderer: path.join(__dirname, 'styleguide/components/StyleGuide'),
  // },
  styles: {
    Code: {
      code: {
        '& .hljs-string': {
          color: 'rgb(43, 132, 13)'
        }
      }
    },
    ComponentsList: {
      item: {
        lineHeight: '1.3'
      }
    },
    Link: {
      link: {
        '&, &:link, &:visited': {
          textDecoration: 'underline'
        },
        '&:hover, &:active': {
          textDecoration: 'none'
        },
        '&:focus': {
          boxShadow: '0 0 0 3px rgb(192, 223, 254)',
          textDecoration: 'none'
        }
      }
    }
  },
  theme: {
    color: {
      base: 'rgb(8, 31, 44)',
      light: 'rgb(111, 111, 111)',
      lightest: 'rgb(216, 216, 216)',
      link: 'rgb(0, 100, 180)',
      linkHover: 'rgb(44, 86, 151)',
      focus: 'rgb(0, 64, 136)',
      border: 'rgb(243, 243, 243)',
      name: 'rgb(43, 132, 13)',
      type: 'rgb(132, 11, 85)',
      error: 'rgb(218, 41, 28)',
      baseBackground: 'rgb(255, 255, 255)',
      codeBackground: 'rgb(254, 254, 254)',
      sidebarBackground: 'rgb(243, 243, 243)',
      ribbonBackground: 'rgb(0, 100, 180)',
      ribbonText: 'rgb(255, 255, 255)',
      codeBase: 'rgb(8, 31, 44)',
      codeComment: 'rgb(111, 111, 111)',
      codePunctuation: 'rgb(118, 118, 118)',
      codeProperty: 'rgb(132, 11, 85)',
      codeDeleted: 'rgb(132, 11, 85)',
      codeString: 'rgb(43, 132, 13)',
      codeInserted: 'rgb(43, 132, 13)',
      codeOperator: 'rgb(85, 71, 65)',
      codeKeyword: 'rgb(105, 31, 130)',
      codeFunction: 'rgb(223, 70, 97)',
      codeVariable: 'rgb(148, 69, 11)'
    },
    fontSize: {
      base: 16,
      text: 16,
      small: 14,
      h1: 48,
      h2: 40,
      h3: 33,
      h4: 28,
      h5: 23,
      h6: 19
    },
    mq: {
      small: '@media (max-width: 719px)'
    },
    borderRadius: 0,
    spaceFactor: 12
  }
};
